frontend/src/components/script-editor.tsx

```diff
@@ -9,23 +9,38 @@ import refreshImage from "@/images/refresh-icon.svg";
 import splitImage from "@/images/split.svg";
 import unsplitImage from "@/images/unsplit.svg";
 import clsx from "@/utils/lib";
-import { useState } from "react";
+import { Dispatch, SetStateAction, useState } from "react";
+import { StackItem } from "../../types";
 
 export default function ScriptEditor() {
   const [scriptSig, setScriptSig] = useState("ScriptSig");
   const [scriptPubKey, setScriptPubKey] = useState("ScriptPubKey");
 
-  const [stackContent, setStackContent] = useState<
-    { id: number; value: string }[]
-  >([]);
+  const [stackContent, setStackContent] = useState<StackItem[]>([]);
+  const [isFetching, setIsFetching] = useState(false);
+  const [error, setError] = useState(null);
 
-  const handleRunScript = () => {
-    const newStackContent = [
-      { id: 1, value: "0x42" },
-      { id: 2, value: "0x01" },
-      { id: 3, value: "0x03" },
-    ];
-    setStackContent(newStackContent);
+  const handleRunScript = async () => {
+    const stack: StackItem[] = [];
```

Comment:
Let's add checks here also to check pub key and sig sizes before even making the request to the backend.

Reply:
Okay boss